Lib/typing.py

```diff
@@ -178,7 +180,11 @@ def _type_check(arg, msg, is_argument=True, module=None, *, allow_special_forms=
 
     We append the repr() of the actual value (truncated to 100 chars).
     """
-    invalid_generic_forms = (Generic, Protocol)
+    try:
+        invalid_generic_forms = (Generic, Protocol)
+    except NameError:
+        # Run during interpreter startup
+        return arg
```

Comment:
There already is a `_typing` module actually, which just defines an identity function used to speed up `NewType` "instantiation".

Reply:
Yes, I'll use that.

Reply:
I added them to the `_typing` module, which is indeed a lot simpler.

A few wrinkles:

- Previously the code was designed to work even if `_typing` was not available. Now that doesn't seem realistic, so I changed it to unconditionally expect the `_typing` module instead.
- I had to add Py_BUILD_CORE to be able to poke at the right bits of the interpreter state. Not great, but I'm not sure there's a better alternative.
- Should I change the name of the classes to be e.g. `_typing.TypeVar`? Probably yes.

Reply:
Seems like that's not enough, because `typing.py` gets imported during deepfreeze, and at that point `_typing` is not yet available. @carljm any ideas on how to adjust the build process to fix that?

Reply:
Fixed by adjusting the Modules/Setup.* files.